packages/flutter_tools/lib/src/build_system/file_hash_store.dart

```diff
@@ -123,7 +123,7 @@ class FileHashStore {
     printTrace('Persisting file store');
     final File file = _cacheFile;
     if (!file.existsSync()) {
-      file.createSync();
+      file.createSync(recursive: true);
```

Comment:
If the parent directory was missing this crashed with a missing file exception. The test added below exercises this case